### PR TITLE
[#817] Quarkus integration

### LINKED ISSUE #817: Add a Quarkus integration to allow running in the JVM and Native mode
### Description
There is a repo with reproducer.
### Actual behavior
dumps NPE short stacktrace to stdout
```console
Caused by: java.lang.NullPointerException
	at com.blazebit.persistence.integration.hibernate.Hibernate53Integrator.integrate(Hibernate53Integrator.java:58)
	at org.hibernate.internal.SessionFactoryImpl.<init>(SessionFactoryImpl.java:275)
	at org.hibernate.boot.internal.SessionFactoryBuilderImpl.build(SessionFactoryBuilderImpl.java:462)
	at io.quarkus.hibernate.orm.runtime.boot.FastBootEntityManagerFactoryBuilder.build(FastBootEntityManagerFactoryBuilder.java:64)
	... 13 more
```
### Steps to reproduce
This commit
https://github.com/soberich/marvel/commit/66e29c94071b75345b132b453c092a12575e8b0f

on this branch [bigfix/blaze-fails-to-integate-with-hibernate](https://github.com/soberich/marvel/commits/bigfix/blaze-fails-to-integate-with-hibernate)
shows how I added `blaze` to try it out with `quarkus`.

```sh
git clone -b bigfix/blaze-fails-to-integate-with-hibernate git@github.com:soberich/marvel.git;
cd marvel;
./gradlew quarkusDev -S
```
On startup there would be an exception.

### Environment

Version:            [1.3.2,1.4.1)
JPA-Provider:       Hibernate 5.4
DBMS:     H2         
Application Server: Netty/Undertow

### PR DESCRIPTION
**Pull request by Mobe91**
fixes #817

### CODE REVIEW

File: entity-view/impl/src/main/java/com/blazebit/persistence/view/impl/tx/TransactionHelper.java [hunk 1]
```diff
@@ -36,6 +40,17 @@
...
     private static volatile TransactionAccessFactory factory;
 
+    private static final String HIBERNATE_IMPLEMENTATION_VERSION;
+
+    static {
+        try {
+            HIBERNATE_IMPLEMENTATION_VERSION = Class.forName("org.hibernate.Session")
+                        .getPackage().getImplementationVersion();
+        } catch (ClassNotFoundException e) {
+            throw new IllegalArgumentException("Unsupported jpa provider!", e);
```
Comment (beikov) [inline entity-view/impl/src/main/java/com/blazebit/persistence/view/impl/tx/TransactionHelper.java:50]: This causes https://travis-ci.com/github/Blazebit/blaze-persistence/jobs/326000071#L3448

IMO this should not cause an exception

Comment (Mobe91) [reply entity-view/impl/src/main/java/com/blazebit/persistence/view/impl/tx/TransactionHelper.java:50]: This is redundant anyway.

File: examples/quarkus/testsuite/native/h2/pom.xml [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+                <artifactId>maven-failsafe-plugin</artifactId>
+                <executions>
+                    <execution>
+                        <goals>
+                            <goal>integration-test</goal>
+                            <goal>verify</goal>
+                        </goals>
+                        <configuration>
+                            <systemProperties>
+                                <native.image.path>${project.build.directory}/${project.build.finalName}-runner.exe</native.image.path>
```
[the remark this answers is not in the document]
Comment (Mobe91) [reply examples/quarkus/testsuite/native/h2/pom.xml:61]: It seems like it does ^^ but it clearly is a relict from my Windows testing..

[the remark this answers is not in the document]
Comment (Mobe91) [reply examples/quarkus/testsuite/native/h2/pom.xml:61]: Do you want me to introduce OS-aware profiles to handles this?

**Pull request merged**